client/faucet/src/mint.rs

```diff
@@ -79,12 +105,28 @@ impl MintParams {
                 self.currency_code(),
                 self.receiver(),
                 self.amount,
-                vec![],
+                self.bcs_metadata(),
                 vec![],
             ),
         )
     }
 
+    fn bcs_metadata(&self) -> Vec<u8> {
+        let trade_id = match self.trade_id.clone() {
+            Some(trade_id) => trade_id,
+            _ => random_uuid(),
+        };
+        let metadata = metadata::Metadata::CoinTradeMetadata(
+            metadata::CoinTradeMetadata::CoinTradeMetadataV0(metadata::CoinTradeMetadataV0 {
+                trade_ids: vec![trade_id],
```

Comment:
@xli and @davidiw This CoinTradeMetadata is new. What's the spec for this? What are valid values for these trade_ids to take? Are they required to be Uuids or just any UTF-8 string?

Reply:
> at least for now

So forever since there's no way to go back :)